contracts/Hub.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.7;
+
+import "./samples/AuthorizedAddresses.sol";
+
+/**
+ * @title Hub Contract that coordinates Storage, Fund and Factory contract.
+ * @dev The contract can mint new synthtic assets (erc20 token), check the collateral ratio of addresses
+ * and liquidate addresses if the collateral ratio falls below 150%
+ * @author The Everest team.
+ */
+
+contract Hub is AuthorizedAddresses {
+    // Contract addresses
+    ICollateralFunds public collateralFundsContract;
+    IStorage public storageContract;
+    IUniswapV3Factory public factory;
+
+    mapping(string => address) public tickersymbolToSynthAssetContractAddress;
+    mapping(string => address) public tickerSymbolToTradingPool;
+    mapping(address => string[]) public userAddressToOpenSynthPositions;
+
+    address public USDCKovan = 0xe22da380ee6B445bb8273C81944ADEB6E8450422;
+    address public uniswapV3Factory = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+
+    /**
+     * @notice gets the synthAsset contract Address from its tickerSymbol
+     * @param _tickerSymbol tickerSymbol to identify the synthAsset
+     * @return _addr address of synthAsset
+     */
+    function getSynthAssetContractAddress (string memory _tickerSymbol) public view returns (address) {
+        return tickersymbolToSynthAssetContractAddress[_tickerSymbol];
+    }
+
+    /**
+     * @notice mints synthAssets to a specific address
+     * @param _receiver address to which token gets minted
+     * @param _amount amount of token that gets minted
+     * @param _tickerSymbol identifier of which token gets minted
+     */
+    function mintSynthAsset(address _receiver, uint256 _amount, string memory _tickerSymbol) public onlyAuthorizedAddresses {
```

Comment:
why `onlyAuthorizedAddresses`, this function is called by users right? and calls synthetic contract `mint()` ?